source/includes/write/bulk-write.c

```diff
@@ -0,0 +1,129 @@
+#include <stdio.h>
+#include <bson/bson.h>
+#include <mongoc/mongoc.h>
+
+int main (void)
+{
+    mongoc_client_t *client;
+    mongoc_collection_t *collection;
+    mongoc_init ();
+
+    client = mongoc_client_new ("<connection string URI>");
+    collection = mongoc_client_get_collection (client, "sample_restaurants", "restaurants");
+
+    // Retrieves a bulk operation handle
+    // start-create-bulk-write
+    mongoc_bulk_operation_t *bulk =
+       mongoc_collection_create_bulk_operation_with_opts (collection, NULL);
+    // end-create-bulk-write
+
+    {
+        // Creates insert operation instructions and adds the operation to the bulk write
+        // start-bulk-insert-one
+        bson_t *insert_doc = BCON_NEW (
+            "name", BCON_UTF8 ("Mongo's Deli"),
+            "cuisine", BCON_UTF8 ("Sandwiches"),
+            "borough", BCON_UTF8 ("Manhattan"),
+            "restaurant_id", BCON_UTF8 ("1234")
+        );
+
+        mongoc_bulk_operation_insert_with_opts (bulk, insert_doc, NULL);
```

Comment:
Add an out-parameter for a `bson_error_t`. Suggest checking the return:

```c
bson_error_t error;
if (!mongoc_bulk_operation_insert_with_opts (bulk, insert_doc, NULL, &error)) {
    fprintf (stderr, "error adding insert operation: %s\n", error.message);
}
```

Suggestion also applies to other `mongoc_bulk_operation_*_with_opts` calls.